libraries/AC_AttitudeControl/AC_PosControl.h

```diff
@@ -582,29 +900,30 @@ class AC_PosControl
+    Vector3f    _accel_target_neu_mss;     // acceleration target in NEU cm/s/s
+    // todo: seperate the limit vector into ne and u. ne is based on acceleration while u is set +-1 based on throttle saturation. Together they don't form a direction vector because the units are different.
+    Vector3f    _limit_vector_neu;         // the direction that the position controller is limited, zero when not limited
 
     // terrain handling variables
-    float    _pos_terrain_target_u_cm;    // position terrain target in cm relative to the EKF origin in NEU frame
-    float    _pos_terrain_u_cm;           // position terrain in cm from the EKF origin in NEU frame.  this terrain moves towards _pos_terrain_target_u_cm
-    float    _vel_terrain_u_cms;          // velocity terrain in NEU cm/s calculated by pos_to_rate step.  this terrain moves towards _vel_terrain_target
-    float    _accel_terrain_u_cmss;       // acceleration terrain in NEU cm/s/s
+    float    _pos_terrain_target_u_m;    // position terrain target in cm relative to the EKF origin in NEU frame
+    float    _pos_terrain_u_m;           // position terrain in cm from the EKF origin in NEU frame.  this terrain moves towards _pos_terrain_target_u_m
+    float    _vel_terrain_u_ms;          // velocity terrain in NEU cm/s calculated by pos_to_rate step.  this terrain moves towards _vel_terrain_target
+    float    _accel_terrain_u_mss;       // acceleration terrain in NEU cm/s/s
 
     // offset handling variables
-    Vector3p    _pos_offset_target_neu_cm;      // position offset target in cm relative to the EKF origin in NEU frame
-    Vector3p    _pos_offset_neu_cm;             // position offset in cm from the EKF origin in NEU frame.  this offset moves towards _pos_offset_target_neu_cm
-    Vector3f    _vel_offset_target_neu_cms;     // velocity offset target in cm/s in NEU frame
-    Vector3f    _vel_offset_neu_cms;            // velocity offset in NEU cm/s calculated by pos_to_rate step.  this offset moves towards _vel_offset_target_neu_cms
-    Vector3f    _accel_offset_target_neu_cmss;  // acceleration offset target in cm/s/s in NEU frame
-    Vector3f    _accel_offset_neu_cmss;         // acceleration offset in NEU cm/s/s
+    Vector3p    _pos_offset_target_neu_m;      // position offset target in cm relative to the EKF origin in NEU frame
+    Vector3p    _pos_offset_neu_m;             // position offset in cm from the EKF origin in NEU frame.  this offset moves towards _pos_offset_target_neu_m
+    Vector3f    _vel_offset_target_neu_ms;     // velocity offset target in cm/s in NEU frame
+    Vector3f    _vel_offset_neu_ms;            // velocity offset in NEU cm/s calculated by pos_to_rate step.  this offset moves towards _vel_offset_target_neu_ms
+    Vector3f    _accel_offset_target_neu_mss;  // acceleration offset target in cm/s/s in NEU frame
+    Vector3f    _accel_offset_neu_mss;         // acceleration offset in NEU cm/s/s
```

Comment:
the comments to the right should be updated from cm/s/s to m/s/s.  same goes for many of the other comments besides the local variables above too